src/legacy/core_plugins/kibana/public/management/sections/index_patterns/edit_index_pattern/create_edit_field/create_edit_field.js

```diff
@@ -17,7 +17,9 @@
  * under the License.
  */
 
-import { Field } from 'ui/index_patterns';
+import { setup as data } from '../../../../../../../data/public/legacy';
+const { FieldImpl: Field } = data.indexPatterns.__LEGACY;
```

Comment:
This is a new pattern right, exporting `__LEGACY`?
It looks great, but should we make it a standard? If not, maybe it's a bit confusing?

Reply:
The convention here isn't new, the naming is just different. This is equivalent to  `__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED` which has been used elsewhere.

(I felt `__LEGACY` was a bit more friendly)  😄

Reply:
This naming is much better, yes.
However, I didn't realize this was to be actually imported from plugins.
What would be the criteria to export something in __LEGACY rather than just export it normally?

Reply:
In an ideal world, the normal contracts we provide in our shim plugins should mirror what will be exposed in the new platform. `__LEGACY` identifies items that are **definitely** going away before new platform and only exist today to support backwards compatibility from the shim plugin. 

So in this case, the class implementation of `Field` needs to be re-exported by `ui/index_patterns` for use in the index patterns management UI, but that's the only place it would be used. Since we haven't yet determined whether index patterns UI should be co-located in the data plugin, I didn't want anybody else to run across this interface and assume it is stable/safe to use as it may become purely internal later. Alternative names could be `__INTERNALS`, `__UNSTABLE` or similar.

Reply:
As discussed offline, we'll keep this as-is for now since we don't otherwise have a way to identify legacy/temporary services that live in runtime contracts. It will probably be refactored out in the very near future anyway.